azure/services/agentpools/spec.go

```diff
@@ -202,16 +213,30 @@ func (s *AgentPoolSpec) Parameters(existing interface{}) (params interface{}, er
 			MaxPods:             s.MaxPods,
 			MinCount:            s.MinCount,
 			Mode:                containerservice.AgentPoolMode(s.Mode),
-			NodeLabels:          s.NodeLabels,
+			NodeLabels:          nodeLabels,
 			NodeTaints:          nodeTaints,
 			OrchestratorVersion: s.Version,
 			OsDiskSizeGB:        &s.OSDiskSizeGB,
 			OsDiskType:          containerservice.OSDiskType(to.String(s.OsDiskType)),
 			OsType:              containerservice.OSType(to.String(s.OSType)),
+			ScaleSetPriority:    containerservice.ScaleSetPriority(to.String(s.ScaleSetPriority)),
 			Type:                containerservice.AgentPoolTypeVirtualMachineScaleSets,
 			VMSize:              sku,
 			VnetSubnetID:        vnetSubnetID,
 			EnableNodePublicIP:  s.EnableNodePublicIP,
 		},
 	}, nil
 }
+
+// mergeSystemNodeLabels appends any kubernetes.azure.com-prefixed labels from the AKS label set
+// into the local capz label set.
+func mergeSystemNodeLabels(capz, aks map[string]*string) map[string]*string {
```

Comment:
does this intentionally drop any labels that are not system labels (ie. no kubernetes.azure.com prefix) but aren't part of the CAPZ spec? for example if they were added by some external process

Reply:
Yes. I think the idea here is that capz is the "exclusive" maintainer of labels, and the update interface has been defined such that every time you push a labels change you need to include the entire set of labels. Ref:

https://learn.microsoft.com/en-us/azure/aks/use-labels#updating-labels-on-existing-node-pools

The think about the `kubernetes.azure.com`-prefixed labels is (IMO) and AKS bug, and hopefully we can drop this foo at some point in the future.

Reply:
https://github.com/Azure/AKS/issues/3152